lib/msal-angular/src/msal.service.ts

```diff
@@ -189,12 +160,24 @@ export class MsalService extends UserAgentApplication {
         }
 
         const protectedResourceMap = frameworkProtectedResourceMap && frameworkProtectedResourceMap.size ? frameworkProtectedResourceMap : new Map(this.msalAngularConfig.protectedResourceMap);
-
-        // process all protected resources and send the matched one
-        const keyForEndpoint = Array.from(protectedResourceMap.keys()).find(key => endpoint.indexOf(key) > -1);
```

Comment:
Applications can provide just the base url of their API in the protected resource map ([example]( https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/2036)), and any API calls with that base url will get tokens attached (since it previously did endpoint.indexOf(baseUrl)). I think we may want to fallback to indexOf  to make sure this is still support, since I don't think minimatch will match that case without the `*`.

Reply:
Done. Test added.